### PR TITLE
Set job_config for babysit task

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
- Set the matching job_config when running babysit task for copr_build.
- Fixes the real cause of https://github.com/packit-service/packit-service/issues/646

TODO:
- [x] Tests
- [x] Check why BuildHelper does not calculate the job configs properly and if we cannot use `get_config_for_handler_kls` for that.
  - `get_config_for_handler_kls` uses `handler` => we can't use it
  - It's better to se correct values `BuildHelper` property `job_build` from outside. (The better solution cal be done after `JobType` cleanup.)

### CODE REVIEW

**Comment by lachmanfrantisek** [comment]
@TomasTomecek I've added some (a little bit lazy) tests, PTAL

File: tests/integration/test_babysit.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                            repo_name="repo_name",
+                            namespace="the-namespace",
+                            project_url="https://github.com/the-namespace/repo_name",
+                        ),
+                        pr_id=5,
+                        job_config_trigger_type=JobConfigTriggerType.pull_request,
+                    )
+                )
+                .mock(),
+            )
```
Comment (TomasTomecek) [inline tests/integration/test_babysit.py:96]: you sure are adventurous, I didn't even know you can do this with flexmock

File: packit_service/worker/jobs.py [hunk 2]
```diff
@@ -142,7 +145,10 @@ def get_config_for_handler_kls(
 
     matching_job_types = MAP_HANDLER_TO_JOB_TYPES[handler_kls]
     for job in jobs_that_can_be_triggered:
-        if job.type in matching_job_types and job not in matching_jobs:
+        if (
+            any(are_job_types_same(job.type, type) for type in matching_job_types)
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py:149]: this is starting to be more and more complicated: if you still have this particular change in your mind, could you please comment in the code here what's happening?

Comment (lachmanfrantisek) [reply packit_service/worker/jobs.py:149]: Still same story about `build` x `copr_build`...;( I'll make a note here.

File: tests/integration/test_babysit.py (hunk 1 above)
Comment (lachmanfrantisek) [reply tests/integration/test_babysit.py:96]: Yes, but I would not consider this as a nice and clean test-case. (But don't want to spend too much time on it now.)

File: packit_service/worker/jobs.py (hunk 2 above)
Comment (lachmanfrantisek) [reply packit_service/worker/jobs.py:149]: > this is starting to be more and more complicated

Sadly, yes. Those aliases and requires don't help us much. I hope we can refactor this once we clear the enums a bit.

Comment (lachmanfrantisek) [reply packit_service/worker/jobs.py:152]: Comment added.

**Comment by lachmanfrantisek** [comment]
The tests should be fixed now. (My global config fixture discovered some problems.)

@TomasTomecek PTAL

**Comment by lachmanfrantisek** [comment]
Last time... Now, it does not check the urls generated by server since it can change in absence/presence of the port.

**Comment by lachmanfrantisek** [comment]
@TomasTomecek finally, the tests are passing.

Can we merge and move the `stable` branch, please?

**Pull request merged**